[mlir][spirv] Add mlir-spirv-tests CI to run for mlir-spv target tests

davidegrohmann opened the pull request:
This should execute also the MLIR SPIRV Target tests which require the SPIRV-Tools validator

davidegrohmann committed:
[mlir][spv] Add mlir-spirv-tests CI to run for mlir-spv target tests

This should execute also the MLIR SPIRV Target tests which require the
SPIRV-Tools validation

Signed-off-by: Davide Grohmann <davide.grohmann@arm.com>
Change-Id: Ied323152e36b95d6f21ed7d4ce4f5f813f280f17

kuhar commented on .github/workflows/spirv-tests.yml:
```diff
@@ -9,6 +9,7 @@ on:
     paths:
       - 'llvm/lib/Target/SPIRV/**'
       - 'llvm/test/CodeGen/SPIRV/**'
+      - 'mlir/test/Target/SPIRV/**'
```
I think we also need the other spirv paths, including dialect definition and conversions

davidegrohmann replied:
Why? They do not generate SPIR-V assembly, so they do not require spirv-tools support. Or am I missing something?

kuhar replied:
Also it's not just Target tests, also the serializer/deserializer implementation

davidegrohmann replied:
Updated

kuhar commented on .github/workflows/spirv-tests.yml:
```diff
@@ -23,7 +24,7 @@ jobs:
-      extra_cmake_args: '-DLLVM_TARGETS_TO_BUILD="SPIRV" -DLLVM_INCLUDE_SPIRV_TOOLS_TESTS=ON'
+      build_target: check-llvm-codegen-spirv check-mlir
+      projects: mlir
+      extra_cmake_args: '-DLLVM_TARGETS_TO_BUILD="X86;SPIRV" -DLLVM_INCLUDE_SPIRV_TOOLS_TESTS=ON'
```
Should this be 'HOST' instead?

davidegrohmann replied:
That does not seem to work:
```
CMake Error at CMakeLists.txt:1116 (message):
  The target `HOST' is not a core tier target.  It may be experimental, if so
  it must be passed via LLVM_EXPERIMENTAL_TARGETS_TO_BUILD.

  Core tier targets:
  AArch64;AMDGPU;ARM;AVR;BPF;Hexagon;Lanai;LoongArch;Mips;MSP430;NVPTX;PowerPC;RISCV;Sparc;SPIRV;SystemZ;VE;WebAssembly;X86;XCore


  Known experimental targets: ARC;CSKY;DirectX;M68k;Xtensa
```

kuhar reviewed (commented):
I'm wondering if it would be better to bifurcate `spirv-tests.yml` and have a separate version for mlir-spirv. It's very rare that we have PRs that affect both implementations, so having them separate could save us some CI time.

davidegrohmann committed:
Change LLVM_TARGETS_TO_BUILD from X86 to host

Signed-off-by: Davide Grohmann <davide.grohmann@arm.com>
Change-Id: I52a5fcf79aba2c272a2ac6a4dc2c393a02b398d9

davidegrohmann committed:
Extend triggering paths

- add SPIRV dialect definition and implementation
- add serialization/deserialization implementation

Signed-off-by: Davide Grohmann <davide.grohmann@arm.com>
Change-Id: Ib63045fec041ba7a2d52b527589c30157ed56258

davidegrohmann committed:
Add one more include path to tge trigger list

Signed-off-by: Davide Grohmann <davide.grohmann@arm.com>
Change-Id: I9919a359c8fbefbc5dd030f9a19c7a83fab12682

davidegrohmann commented:
> MLIR SPIR-V tests seem to take ~1h (https://github.com/llvm/llvm-project/actions/workflows/mlir-spirv-tests.yml) compared to regular checks that finish in ~5-10min. I wonder if there is any reason for that and whether we can bring the time down?

I can have a look. Can you point me to the workflows scripts for the regular checks so I can compare those with the mlir spir-v tests script?

IgWod-IMG commented:
I have also noticed that auto-merge doesn't wait for this job: https://github.com/llvm/llvm-project/pull/153440 We also should address that. Not sure if this PR is the best place to track those issues.
